design/one-pager-resource-status.md

```diff
@@ -0,0 +1,133 @@
+# Resource High Level Stop-Light Status
+
+- Owner: Steven Rathbauer ([@rathpc](https://github.com/rathpc))
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Problem
+
+As a stack author there is currently no way to author rules that combine arbitrary conditions of a given state to
+represent high level _"stop-light"_ conditions.
+
+The high level conditions would be:
+- Online
+- Offline
+- Warning
+- Unknown
+
+The **Unknown** condition is interesting because this would be returned if the conditions were annotated but we cannot
+verify a known state based on the given rules. _(Essentially a default case in a switch block)_.
+
+## Design
+
+In contrast to the design [here](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-stack-status-spec.md),
+this design is meant to handle combinations of arbitrary statuses to determine a high level "overall" status.
+
+The following examples are meant to be added to a `ui-schema.yaml` file. The `resourceStatus` key should be added as a root key in that file.
+
+<a name="example-1"></a>
+
+```yaml
+# Example 1
+resourceStatus:
+  paths: .status.conditionedStatus.conditions
+  states:
+    Online:
+    - Ready: 'True'
+      Synchronized: 'True'
+    Offline:
+    - Ready: 'False'
+      Synchronized: 'False'
+    Warning:
+    - Ready: 'False'
+      Synchronized: 'True'
+      SyncFailure: 'True'
+```
+
+<a name="example-2"></a>
+
+```yaml
+# Example 2
+resourceStatus:
+  paths:
```

Comment:
I am concerned about the parsing complexities of having `paths` be either a string or a map, mirrored here https://github.com/crossplaneio/crossplane/pull/1145#discussion_r362497437.

Perhaps `generatePaths` or `customPaths` could be used as a name for the map structure.

Reply:
As per a discussion we had, I think the best short term solution to this problem is to establish that the ui-schema file is explicitly formatted per OpenAPI specs which allows for "OR" conditions. This will also allow us to lint (to a degree) this file to prevent issues after publishing due to incorrectly formatted files.